Share button that opens a Share Intent for PodcastDetailActivity

jaketothepast opened the pull request:
Implemented a button that opens a share intent in the Podcast Detail Activity with the text "Check out this episode of Software Engineering Daily: <post link>."
  
Fixes #15

jaketothepast committed:
Pitiful attempt to start a button for a share intent

jaketothepast committed:
Starting a basic sharing intent

jaketothepast committed:
Making the layout for the share button nicer, hopefully finished up the share intent

jaketothepast committed:
Minor wording change, added a return statement

jaketothepast commented:
Ready to make any changes you need!

TheHollidayInn commented:
Looks great! Do you mind adding a screenshot of the detail activity to the description of this PR?

jaketothepast commented:
Of course, here is how the activity renders on my phone complete with a great pic of Stephen Wolfram
![screenshot_20180107-213019](https://user-images.githubusercontent.com/26175457/34657284-a099fa5a-f3f2-11e7-997c-abde09d40a14.png)

TheHollidayInn commented:
Thanks! How about adding the share button to the actionbar along with the share icon: https://developer.android.com/training/sharing/shareaction.html

jaketothepast commented:
Yep I'll get right on that

jaketothepast committed:
Starting to add this to the app bar

jaketothepast commented:
Back on it, been a crazy week at work. Sorry in advance for the nooby question, really trying to get my feet wet with Android here, is the actionbar in this context the collapsing toolbar view on the detail activity?

jaketothepast committed:
Created a menu to hopefully soon render in the action bar

jaketothepast committed:
Import menu, add a menuinflater to the options menu for the detail activitiy

jaketothepast committed:
Remove old bad button, add share icon to menu item

jaketothepast commented:
@TheHollidayInn How does this look?

![screenshot_1516077016](https://user-images.githubusercontent.com/26175457/34971936-1a52a23e-fa4c-11e7-93da-4464d72f5077.png)
![screenshot_1516077066](https://user-images.githubusercontent.com/26175457/34971948-33d767f8-fa4c-11e7-886f-f4ec56823bff.png)

TheHollidayInn commented:
Ah, I'm sorry. I missed your message. This does look much better! However, the toolbar is the action bar in this context. It looks like you have thing correct, I am going to leave a few more comments.

TheHollidayInn commented on app/src/main/res/layout/activity_podcast_detail.xml:
```diff
@@ -162,17 +173,29 @@
                     android:textSize="14sp"
                     android:textStyle="bold" />
+
+                <Button
```
What does this button do?

TheHollidayInn commented on app/src/main/res/layout/activity_podcast_detail.xml:
```diff
@@ -139,21 +150,21 @@
             <RelativeLayout
                 android:layout_width="match_parent"
-                android:layout_height="match_parent"
+                android:layout_height="194dp"
```
Why did you restrict the height of this rather than let it match the parent?

TheHollidayInn commented on app/src/main/res/layout/activity_podcast_detail.xml:
```diff
@@ -89,6 +89,17 @@
+                    android:layout_marginBottom="10dp"
+                    android:layout_marginEnd="10dp"
+                    android:layout_marginRight="10dp"
+                    android:contentDescription="Share this Episode"
```
Can you add these strings to the string resource and use their `@string/`

TheHollidayInn reviewed (changes requested):
Looks good! Almost there :D

jaketothepast committed:
Actually add the share intent to the icon

jaketothepast committed:
Remove old share button, add padding to toolbar

jaketothepast commented:
Ok think at this point @TheHollidayInn we might be done or at least nearly finished. I ended up removing the original share button I added during my first baby steps with Android since now we have the share icon. Also I removed that old button, and changed the layoutWidth back to match parent.

While making those changes I realized I hadn't properly added the share intent to the icon I created, so I added some code for that too.

TheHollidayInn commented:
Alright! Looking great :D.

I'm working on a few things this week and will be sure to get this merged in before the next release.
